pkg/build/builder/common.go

```diff
@@ -180,3 +155,54 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateBuildRevision(build *api.Build, sourceInfo *git.SourceInfo) *api.SourceRevision {
+	if build.Spec.Revision != nil {
+		return build.Spec.Revision
+	}
+	return &api.SourceRevision{
+		Git: &api.GitSourceRevision{
+			Commit:  sourceInfo.CommitID,
+			Message: sourceInfo.Message,
+			Author: api.SourceControlUser{
+				Name:  sourceInfo.AuthorName,
+				Email: sourceInfo.AuthorEmail,
+			},
+			Committer: api.SourceControlUser{
+				Name:  sourceInfo.CommitterName,
+				Email: sourceInfo.CommitterEmail,
+			},
+		},
+	}
+}
+
+type buildDetails struct {
+	revision *api.SourceRevision
+	status   *api.BuildStatus
+}
```

Comment:
@bparees  there are a few ways i thought about updating the build revision of the build object together with the failure reason and message.
the first one is duplicating the updateStatusRetrier function below and making one specific for updating the revision. which would be very little different than the one in place now. (just set rev and reset resourceversion)

the second is changing the function signature and adding separate details for revision and failure status and message, like 
```
func updateStatusRetrier(buildName string, client client.BuildInterface, revision *api.ResourceRevision, reason api.StatusReason, message string) error 
```
the downside of the above is that every time i don't have a revision, which is a lot, i would end up initializing a pointer with no value to the function and it just seems inappropriate to me. plus i no longer look for a nil pointer, but for an empty field in the struct.
```
updateStatusRetrier(build.Name, client, &api.ResourceRevision{}, reason, message)
```

and the third option is a struct, like you can see here that can also accommodate the build substatus as well. the downside of this being that it looks very complicated and it makes me feel like i'm trying to engineer something where there's no need. and it also is verbose where i use it, see the code down below. (and hard to understand, maybe?)

so my question is, which do i pick? is there a forth option?